packages/xstate-test/src/machine.ts

```diff
@@ -104,10 +109,15 @@ export function createTestModel<TMachine extends AnyStateMachine>(
   const testModel = new TestModel<StateFrom<TMachine>, EventFrom<TMachine>>(
     machine as SimpleBehavior<any, any>,
     {
-      serializeState: (state, event) => {
-        return `${serializeState(state)}${serializeTransition(state, event, {
-          serializeEvent
-        })}` as SerializedState;
+      serializeState: (state, event, prevState) => {
```

Comment:
To do that, we need `serializeState` in `@xstate/graph` to also consider/provide the previous state (`prevState`).

Fun fact: we can't use `state.history` for this because that's actually deleted for optimization reasons, and `state.history` is deprecated in XState v5 anyway.